migration/run.js

```diff
@@ -1,29 +1,88 @@
-const hasUntilDestroy = /import\s*{\s*[^}]*untilDestroyed[^}]*}\s*from\s*("|')ngx-take-until-destroy\1(?=[^]*untilDestroyed\(\w*\)[^]*)/;
-const catchImport = /import\s*{\s*[^}]*untilDestroyed[^}]*}\s*from\s*("|')ngx-take-until-destroy['|"];/;
+const hasUntilDestroy = /import\s*{\s*[^}]*untilDestroyed[^}]*}\s*from\s*(["'])ngx-take-until-destroy\1(?=[^]*untilDestroyed\(\w*\)[^]*)/;
 
 const glob = require('glob');
 const fs = require('fs');
+const { Project } = require('ts-morph');
 
-const base = `src/app`;
+const base = `app`;
+
+glob(`${base}/**/*.ts`, {}, function(_, files) {
+  const removeOnDestroy = process.argv.includes('--removeOnDestroy');
 
-glob(`${base}/**/*.ts`, {}, function(er, files) {
   files.forEach(path => {
-    fs.readFile(path, 'utf8', function(err, text) {
-      if (hasUntilDestroy.test(text)) {
+    fs.readFile(path, 'utf8', function(_, text) {
+      if (!hasUntilDestroy.test(text)) return;
+
+      const result = transformCode(text, { removeOnDestroy });
+
+      fs.writeFile(path, result, 'utf8', function(err) {
+        if (err) return console.log(err);
         console.log(`Replaced ${path}`);
-        const result = text
-          .replace(
-            /((?:@\w*\([^]*\)[\n\s\r\t]*)(?=([\n\s\r\t]*export[\s\r\t]*class))\2)/,
-            '@UntilDestroy()\n$1'
-          )
-          .replace(
-            catchImport,
-            `import { untilDestroyed, UntilDestroy } from '@ngneat/until-destroy';`
-          );
-        fs.writeFile(path, result, 'utf8', function(err) {
-          if (err) return console.log(err);
-        });
-      }
+      });
     });
   });
 });
+
+function transformCode(str, { removeOnDestroy = false } = {}) {
+  const sourceFile = new Project().createSourceFile(`code.ts`, str);
+  replaceOldImport(sourceFile);
+
+  const classes = sourceFile.getClasses() || [];
+  classes.forEach(clazz => {
+    addUntilDestroyDecorator(clazz);
+
+    if (removeOnDestroy) {
+      const ngOnDestroyMember = clazz.getMember('ngOnDestroy');
+      const isNgOnDestroyEmpty = Boolean(
```

Comment:
```suggestion
      const ngOnDestroyIsNotEmpty = Boolean(
```
You've probably reverted the meaning :stuck_out_tongue_closed_eyes: